### DISCUSSION TITLE
Tags for Performerns

### DISCUSSION THREAD

**Discussion by Tscherno** [stashapp/stash#2002] [Ideas]
Hi,

not sure if the idea already was discussed: Why not add tags to performers? They should go to all of their scenens automatically then (they always are big b**bed, or eye color blue...).

**Comment by SmallCoccinelle** [comment]
We already have tags for performers in the code base. Go `Performers -> PERFORMER -> Edit -> scroll down to "Tags"`.

If you then go to scenes, you can add a filter (looks like a funnel icon) and select performer tags to include them. This will filter your search for performers with certain tags.

Adding the tags into scenes automatically is probably something that should happen in a derived manner. That is, you keep the data separate, but when you *search* for a tag, and there's a performer in the scene with said tag, the scene should occur in the search results. I'm currently experimenting with a search engine, and this is exactly the kind of query it is able to process efficiently, because it can score the occurrence of a performer tag in a scene.

The reason it's desirable to keep performer tags split from scene tags is that it is far easier to "glue" tags together then it is to split them later on. Also, updating a tag set on a performer is easier if you don't have to go and find all scenes where you added it.

Eye-color exist as performer-metadata, and to a certain extent, also measurements. So they tend to live better as meta-data on performers rather than tags. This allows a search query such as `performer.eye-color:blue` or just `blue`, as it will find its way and match on eye-color anyway.
